packages/core/angular/src/wallet/local-storage.ts

```diff
@@ -1,37 +1,32 @@
-export class LocalStorageService<T> {
-    private _value = this._defaultValue;
+import { BehaviorSubject } from 'rxjs';
 
-    constructor(private _key: string, private _defaultValue: T) {}
+export class LocalStorageService<T> {
+    private _value = new BehaviorSubject(this.getInitialValue());
+    value$ = this._value.asObservable();
 
-    get value(): T {
-        if (this._value) return this._value;
+    constructor(private _key: string, private _defaultState: T) {}
 
-        try {
-            const value = localStorage.getItem(this._key);
-            if (value) {
-                this._value = JSON.parse(value) as T;
-                return this._value;
-            }
-        } catch (error) {
-            console.error(error);
+    private getInitialValue(): T {
+        if (typeof localStorage === 'undefined') {
+            return this._defaultState;
         }
 
-        return this._defaultValue;
+        const value = localStorage.getItem(this._key);
+
+        return value ? (JSON.parse(value) as T) : this._defaultState;
     }
 
     setItem(newValue: T): void {
-        if (newValue === this.value) return;
-
-        this._value = newValue;
-
-        try {
-            if (newValue === null) {
-                localStorage.removeItem(this._key);
-            } else {
-                localStorage.setItem(this._key, JSON.stringify(newValue));
-            }
-        } catch (error) {
-            console.error(error);
+        const value = this._value.getValue();
+
+        if (newValue === value) return;
+
+        this._value.next(newValue);
+
+        if (newValue === null) {
+            localStorage.removeItem(this._key);
+        } else {
+            localStorage.setItem(this._key, JSON.stringify(newValue));
```

Comment:
This try/catch block should be left as is, I think. This was a fix made in https://github.com/solana-labs/wallet-adapter/pull/199. Same situation in the getter above.

Reply:
Got it